libecs/source/Core/Event/manager.cxx

```diff
@@ -0,0 +1,16 @@
+#include "pivot/ecs/Core/Event/manager.hxx"
+#include "pivot/ecs/Core/Event/index.hxx"
+
+namespace pivot::ecs::event
+{
+Manager::Manager(systems::Manager &systemManager): m_systemManager(systemManager) {}
+
+void Manager::sendEvent(const std::string &eventName, const data::Value &payload, const std::vector<Entity> &entities)
+{
+    const auto description = GlobalIndex::getSingleton().getDescription(eventName).value();
```

Comment:
Please register the events in the event manager, store their descriptions, and check with the stored description if the value matches the type. The global index is shared between every scene, so we cannot depend on it to get the list of events of this specific scene.